### ISSUE TITLE
3D rotation repeats Z axis

### ISSUE THREAD

**Issue by jrubin01** [biomedia-mira/istn#6]
In def affine_matrix(self, x) for the STN3D class (line 330), the rot Z code block is repeated:

```
# self.rot = self.rotation(x)
rot = torch.tanh(self.rotation(x)) * (math.pi / 4.0)
# rot Z
angle_1 = rot[:, 0].view(-1)
rotation_matrix_1 = torch.zeros([b, 4, 4], dtype=torch.float)
rotation_matrix_1[:, 0, 0] = torch.cos(angle_1)
rotation_matrix_1[:, 0, 1] = -torch.sin(angle_1)
rotation_matrix_1[:, 1, 0] = torch.sin(angle_1)
rotation_matrix_1[:, 1, 1] = torch.cos(angle_1)
rotation_matrix_1[:, 2, 2] = 1.0
rotation_matrix_1[:, 3, 3] = 1.0
# rot X
angle_2 = rot[:, 1].view(-1)
rotation_matrix_2 = torch.zeros([b, 4, 4], dtype=torch.float)
rotation_matrix_2[:, 1, 1] = torch.cos(angle_2)
rotation_matrix_2[:, 1, 2] = -torch.sin(angle_2)
rotation_matrix_2[:, 2, 1] = torch.sin(angle_2)
rotation_matrix_2[:, 2, 2] = torch.cos(angle_2)
rotation_matrix_2[:, 0, 0] = 1.0
rotation_matrix_2[:, 3, 3] = 1.0
# rot Z
angle_3 = rot[:, 2].view(-1)
rotation_matrix_3 = torch.zeros([b, 4, 4], dtype=torch.float)
rotation_matrix_3[:, 0, 0] = torch.cos(angle_3)
rotation_matrix_3[:, 0, 1] = -torch.sin(angle_3)
rotation_matrix_3[:, 1, 0] = torch.sin(angle_3)
rotation_matrix_3[:, 1, 1] = torch.cos(angle_3)
rotation_matrix_3[:, 2, 2] = 1.0
rotation_matrix_3[:, 3, 3] = 1.0
```

Should the last operation rotate around the Y axis instead?

```
# rot Y
angle_3 = rot[:, 2].view(-1)
rotation_matrix_3 = torch.zeros([b, 4, 4], dtype=torch.float)
rotation_matrix_3[:, 0, 0] = torch.cos(angle_3)
rotation_matrix_3[:, 0, 2] = torch.sin(angle_3)
rotation_matrix_3[:, 2, 0] = -torch.sin(angle_3)
rotation_matrix_3[:, 2, 2] = torch.cos(angle_3)
rotation_matrix_3[:, 1, 1] = 1.0
rotation_matrix_3[:, 3, 3] = 1.0
```

**Comment by bglocker** [comment]
We use a [ZXZ convention](https://en.wikipedia.org/wiki/Euler_angles) for 3D rotations with Euler angles. The code should be correct as is.

**Comment by jrubin01** [comment]
Got it. Thanks.
